modules/regex/regex.cpp

```diff
@@ -343,15 +343,14 @@ String RegEx::sub(const String &p_subject, const String &p_replacement, bool p_a
 
 	ERR_FAIL_COND_V(!is_valid(), String());
 
-	String output;
-	output.resize(p_subject.length());
+	PCRE2_SIZE olength = p_subject.length() + 1; // string and terminating \0
+	Vector<CharType> output;
+	output.resize(olength + 1);
```

Comment:
Might be worth adding another comment here to explain why there is another +1, since it ends up as `p_subject.length() + 2`.

Reply:
Added `safety_zone` with a comment to explain what's the idea.